server.py

```diff
@@ -86,30 +114,61 @@ def b64dec(s):
 
 
 def get_verified_email(token):
+    # Request the provider configuration document from Portier. You should
+    # normally cache this for some time, if possible according to max-age in
+    # the Cache-Control header.
     rsp = request.urlopen(
         '%s/.well-known/openid-configuration' % META['portier_origin']
     )
     config = json.loads(rsp.read().decode('utf-8'))
+
+    # Get the keys configuration URL.
     if 'jwks_uri' not in config:
         return {'error': 'No jwks_uri in discovery document.'}
 
+    # Request the current key set. You should cache this as well.
     rsp = request.urlopen(config['jwks_uri'])
     try:
         keys = json.loads(rsp.read().decode('utf-8'))['keys']
     except Exception:
         return {'error': 'Problem finding keys in JWK key set.'}
 
+    # Parse the header off the token, which contains the key ID used to sign
+    # the token, and look for the matching key in the key set.
     raw_header = token.split('.', 1)[0]
     header = json.loads(b64dec(raw_header).decode('utf-8'))
     try:
         key = [k for k in keys if k['kid'] == header['kid']][0]
     except Exception:
         return {'error': 'Cannot find key with ID %s.' % header['kid']}
 
+    # Create an RSAPublicKey instance from the key we found.
     e = int.from_bytes(b64dec(key['e']), 'big')
     n = int.from_bytes(b64dec(key['n']), 'big')
     pub_key = rsa.RSAPublicNumbers(e, n).public_key(default_backend())
 
+    # Verify the token. We let the pyjwt do a lot of work for us, but you
```

Comment:
Adding this after you just removed something similar a few commits back feels pretty silly. Maybe do it in a smarter way, morphing the previous comment into what you want here? (Still feels more useful to provide high-level stuff in the docstring, as it makes it more accessible than just having a comment.)